docs/rfcs/1011-non-blocking-invite-transport.md

```diff
@@ -0,0 +1,1226 @@
+                                "type": "Bytes"
+                            }
+                        ]
+                    },
+                    {
+                        "name": "WaitPeerAcknowledgment",
+                        "discriminant_value": "8"
+                    }
+                ]
+            }
+```
+
+And the `ClaimerStep` nested data type has the following definition:
+
+```json5
+            {
+                "name": "ClaimerStep",
+                "discriminant_field": "step",
+                "variants": [
+                    {
+                        "name": "WaitPeer",
+                        "discriminant_value": "0",
+                        "fields": [
+                            {
+                                "name": "public_key",
+                                "type": "PublicKey"
+                            }
+                        ]
+                    },
+                    {
+                        "name": "SendHashedNonce",
+                        "discriminant_value": "1",
+                        "fields": [
+                            {
+                                "name": "hashed_nonce",
+                                "type": "HashDigest"
+                            }
+                        ]
+                    },
+                    {
+                        "name": "GetNonce",
+                        "discriminant_value": "2"
+                    },
+                    {
+                        "name": "SendNonce",
+                        "discriminant_value": "3",
+                        "fields": [
+                            {
+                                "name": "claimer_nonce",
+                                "type": "Bytes"
+                            }
+                        ]
+                    },
+                    {
+                        "name": "SignifyTrust",
+                        "discriminant_value": "4"
+                    },
+                    {
+                        "name": "WaitPeerTrust",
+                        "discriminant_value": "5"
+                    },
+                    {
+                        "name": "SendPayload",
+                        "discriminant_value": "6",
+                        "fields": [
+                            {
+                                "name": "claimer_payload",
+                                "type": "Bytes"
+                            }
+                        ]
+                    },
+                    {
+                        "name": "GetPayload",
+                        "discriminant_value": "7"
+                    },
+                    {
+                        "name": "Acknowledge",
+                        "discriminant_value": "8"
+                    }
+                ]
+            }
+```
+
+## Typical workflow
+
+Typical workflow
+----------------
+
+Here's how a typical workflow for a user invitation can play out:
+
+```python
```

Comment:
Are you interested by a Sequence Diagram ? you could do that using the mermaid syntax

https://mermaid.js.org/syntax/sequenceDiagram.html

Reply:
Right I'll try that :+1:

Reply:
I don't mind adding diagram, but I think the current pseudo code snippet is very convenient to read in its current form ;-)